tasks/build/download_node_builds.js

```diff
@@ -1,84 +1,134 @@
-          .on('end', cb);
-        });
+  const getNodeBuild = (platform) => {
+    const finalDir = platform.nodeDir;
+    const downloadDir = `${finalDir}.temp`;
+    const file = basename(platform.nodeUrl);
+
+    return wreckGet(platform.nodeUrl)
+    .then(([resp, payload]) => {
+      if (resp.statusCode !== 200) {
+        throw new Error(`${platform.nodeUrl} failed with a ${resp.statusCode} response`);
       }
+      const bytes = parseInt(resp.headers['content-length'], 10) || 'unknown number of';
+      const mb = ((bytes / 1024) / 1024).toFixed(2);
+      grunt.log.ok(`downloading ${platform.name} - ${mb} mb`);
+      grunt.file.mkdir(downloadDir);
+      return payload;
+    })
+    .then(payload => {
+      return writeFileAsync(resolve(downloadDir, file), payload);
+    })
+    .then(() => {
+      return renameAsync(downloadDir, finalDir);
+    });
 
-      await fromNode(cb => {
-        rename(downloadDir, finalDir, cb);
+  };
+
+  const extractNodeBuild = async (platform) => {
+    const file = basename(platform.nodeUrl);
+    const path = resolve(platform.nodeDir, file);
+
+    await fromNode(cb => {
+      createReadStream(path)
+      .pipe(createGunzip())
+      .on('error', (err) => {
+        cb(new Error(`error extracting: ${err.message}`));
+      })
+      .pipe(new Extract({path: platform.nodeDir, strip: 1}))
+      .on('error', (err) => {
+        cb(new Error(`error extracting: ${err.message}`));
+      })
+      .on('end', () => {
+        cb(null, true);
       });
-    }());
+    });
+  };
 
-    activeDownloads.push(platform);
+  let downloadFailed = false;
+  const shouldDownload = async (platform, downloadCounter) => {
+    if (downloadCounter === downloadLimit) {
+      downloadFailed = true;
+      return false;
+    }
+    if (!grunt.file.isDir(platform.nodeDir)) {
+      return true;
+    }
+    return !(await checkShaSum(platform));
+  };
 
-    var bytes = parseInt(resp.headers['content-length'], 10) || 'unknown number of';
-    var mb = ((bytes / 1024) / 1024).toFixed(2);
-    grunt.log.ok(`downloading ${platform.name} - ${mb} mb`);
+  const start = async (platform) => {
+    let downloadCounter = 0;
+    while (await shouldDownload(platform, downloadCounter++)) {
+      if (grunt.file.isDir(platform.nodeDir)) {
+        grunt.file.delete(platform.nodeDir);
+      }
+      await getNodeBuild(platform);
+    }
+
+    if (!platform.win) {
+      await extractNodeBuild(platform);
+    }
+
+    if (downloadFailed) grunt.log.fail(`${platform.name} download failed`);
+    else grunt.log.ok(`${platform.name} download complete`);
   };
 
-  grunt.registerTask('_build:downloadNodeBuilds:start', function () {
-    map(platforms, start).nodeify(this.async());
+  grunt.registerTask('_build:downloadNodeBuilds', function () {
+    const cb = this.async();
+    getShaSums().then(() => {
+      map(platforms, start).nodeify(cb);
+    }).catch(e => {
+      grunt.log.fail(`something went wrong: ${e.message}`);
+    });
```

Comment:
I'd suggest modifying this slightly:

```
const done = this.async();

getShaSums()
.then(() => {
  return map(platforms, start);
})
.nodeify(done);
```

Putting nodeify at the end of the chain will catch any errors and pass them to the grunt async callback. Also I'd stick with the grunt convention of naming the callback `done`. Also, very minor point, but note the indentation. Based on our styleguide, chained methods should be indented at the same level as the first expression in the chain.